### DISCUSSION TITLE
How can I get the attributes of span?

### DISCUSSION THREAD

**Discussion by JooKS-me** [open-telemetry/opentelemetry-java#3376] [Q&A] [answered]
Hi, community~

In my unit test, I need to assert the attributes of span. But, how to get the attributes from the span troubles me a lot.

More specifically:
```java
    @Before
    public void setup() {
        SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
                .addSpanProcessor(BatchSpanProcessor.builder(ZipkinSpanExporter.builder().build()).build())
                .build();
        OpenTelemetrySdk.builder()
                .setTracerProvider(sdkTracerProvider)
                .buildAndRegisterGlobal();
    }

   @Test
    public void assertMethod() {
        // execute method
        // ...
        Span span = Span.current();
        // How can I get attributes here?
    }
```
Are there any APIs that can be used here? Or is there a better way to test?

**Comment by kenfinnigan** [comment, marked as answer]
I usually use the `InMemorySpanExporter` from https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/testing to capture the traces/spans and then retrieve them from the exporter for verification of Span attributes, etc

**Reply by JooKS-me** [reply]
Thanks for your answer, and that looks useful.
